backends/ClimaCommsMPI/src/ClimaCommsMPI.jl

```diff
@@ -147,20 +140,10 @@ function ClimaComms.start(
     end
 end
 
-function ClimaComms.progress(ghost::MPISendRecvGraphContext)
+ClimaComms.progress(ghost::MPISendRecvGraphContext) =
     MPI.Iprobe(MPI.MPI_ANY_SOURCE, ghost.tag, ghost.ctx.mpicomm)
-end
-
-function ClimaComms.finish(
-    ghost::MPISendRecvGraphContext;
-    dependencies = nothing,
-)
-    # wait on previous receives
-    MPI.Waitall!(ghost.recv_reqs)
-    # ensure that sends have completed
-    # TODO: these could be moved to start()? but we would need to add a finalizer to make sure they complete.
```

Comment:
I was actually wondering if it would be better to move the `Waitall` on the sends to the start of the next pass?